pkg/endpoint/endpoint.go

```diff
@@ -1219,11 +1219,14 @@ func (e *Endpoint) GetIdentity() identityPkg.NumericIdentity {
 	return identityPkg.InvalidIdentity
 }
 
-func (e *Endpoint) directoryPath() string {
+// DirectoryPath returns the directory name for this endpoint bpf program.
+func (e *Endpoint) DirectoryPath() string {
 	return filepath.Join(".", fmt.Sprintf("%d", e.ID))
 }
 
-func (e *Endpoint) failedDirectoryPath() string {
+// FailedDirectoryPath returns the directory name for this endpoint bpf program
+// failed builds.
+func (e *Endpoint) FailedDirectoryPath() string {
```

Comment:
Does this now need a read-lock? :(
e.ID never changes so maybe we can just say that and leave a comment on e.ID saying don't ever change it or something (it feels a little silly to lock on this)

Reply:
oh, I see `mutex` is below `ID` in the class definition. I think our style means that it isn't guarded by the mutex?